src/Host/Protocol/Test/UserProfileServicePipe/UserProfileServicePipeTest.cs

```diff
@@ -0,0 +1,135 @@
+        private int TestClientTimeoutMS => 5000;
+
+        [CompositeTest]
+        // valid test data
+        [InlineData("{\"Username\":\"testname\", \"Domain\":\"testdomain\", \"Password\":\"testPassword\"}", "testname", "testdomain", "testPassword", true, true, false)]
+        // Missing quote
+        [InlineData("{Username\":\"testname\", \"Domain\":\"testdomain\", \"Password\":\"testPassword\"}", null, null, null, false, false, false)]
+        // Missing closing paren
+        [InlineData("{", null, null, null, false, false, false)]
+        // empty json object
+        [InlineData("{}", null, null, null, false, false, false)]
+        // No username, domain and password
+        [InlineData("{\"Username\":, \"Domain\":, \"Password\":}", null, null, null, false, false, false)]
+        // empty username domain password
+        [InlineData("{\"Username\": \"\", \"Domain\": \"\", \"Password\": \"\"}", "", "", "", true, false, false)]
+        // whitespace input string
+        [InlineData("                     ", null, null, null, true, false, false)]
+        public async Task CreateProfileTest(string input, string username, string domain, string password, bool isValidParse, bool isValidAccount, bool isExistingAccount) {
+            var creator = UserProfileCreatorMock.Create(username, domain, password, isValidParse, isValidAccount, isExistingAccount);
+            await CreateProfileTestRunnerAsync(creator, input, isValidParse, isValidAccount, isExistingAccount);
+        }
+
+        [Test]
+        public async Task CreateProfileFuzzTest() {
+            string inner = "\"Username\": {0}, \"Domain\": {1}, \"Password\":{2}";
+            List<string> testResults = new List<string>();
+            for (int i = 0; i < 100000; ++i) {
+
+                byte[] usernameBytes = GenerateBytes();
+                byte[] domainBytes = GenerateBytes();
+                byte[] passwordBytes = GenerateBytes();
+
+                string username = Encoding.Unicode.GetString(usernameBytes);
+                string domain = Encoding.Unicode.GetString(domainBytes);
+                string password = Encoding.Unicode.GetString(passwordBytes);
+
+                string json = "{" + string.Format(inner, username, domain, password) + "}";
+                
+                string testResult = string.Empty;
+                UserProfileCreatorFuzzTestMock creator = new UserProfileCreatorFuzzTestMock();
+
+                try {
+                    await CreateProfileTestRunnerAsync(creator, json, false, false, false, true);
+                } catch (IOException) {
+                    // expect pipe to fail. The client side pipe throws an IOException when the server side pipe 
+                    // closes due to IO error or attempt to access unauthorized memory.
+                }
+            }
+        }
+
+        private byte[] GenerateBytes() {
+            Random rd = new Random((int)DateTime.Now.Ticks);
```

Comment:
You can just use the default constructor for `Random` - it automatically uses ticks or something similar.